### PR TITLE
Tweak ESLint config to check .js files

### PR DESCRIPTION
**Pull request by pdehaan**
## What does this PR do?

Ensures that ESLint checks .js files in addition to .json files.

## How do I test this PR?

`npm run lint` (or whatever your fancy yarn equivalent would be).

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -3,7 +3,9 @@
         "es6": true,
         "node": true
     },
-    "extends": "eslint:recommended",
+    "extends": [
+      "eslint:recommended"
+    ],
```
Comment (pdehaan) [inline .eslintrc.json:8]: Converted this to an array syntax so it'd be easier to do a quick "plugin:promise/recommended" if you want to extend that recommended config in the future.

File: .eslintrc.json [hunk 2]
```diff
@@ -12,9 +14,7 @@
       "json"
     ],
     "rules": {
-        "indent": ["error",
-            2
-        ],
+        "indent": ["error", 2],
```
Comment (pdehaan) [inline .eslintrc.json:17]: one-lined all these so it'd be easier to sort rules A-Z. I had a real heck of a time trying to quickly find rules in the unsorted list. But sorting everything in this PR got a bit too noisy.

File: client/coral-admin/src/routes/Community/components/FlaggedUser.js [hunk 3]
```diff
@@ -85,7 +85,7 @@ class User extends React.Component {
                 <span className={styles.flaggedByLabel}>
                   {t('community.flags')}({ user.actions.length })
                 </span>:
-                  { user.action_summaries.map(
+                { user.action_summaries.map(
```
Comment (pdehaan) [inline client/coral-admin/src/routes/Community/components/FlaggedUser.js]: These were auto-fixed `indent` tweaks from `npm run lint-fix`.
Speaking of formatting and fixing, you may want to consider [**prettier**](http://npm.im/prettier) for optimal code formatting.

File: graph/loaders/users.js [hunk 4]
```diff
@@ -27,7 +27,7 @@ const genUserByIDs = async (context, ids) => {
  * @param  {Object} context   graph context
  * @param  {Object} query     query terms to apply to the users query
  */
-const getUsersByQuery = async ({user, loaders: {Actions}}, {ids, limit, cursor, statuses, action_type, sortOrder}) => {
+const getUsersByQuery = async ({loaders: {Actions}}, {ids, limit, cursor, statuses, action_type, sortOrder}) => {
```
Comment (pdehaan) [inline graph/loaders/users.js]: Removed `user` since it seemed unused, and was causing ESLint failures.

**Comment by wyattjoh** [comment]
Looks great to me @pdehaan! I'm merging this into another branch because I had already started on fixing those linting errors with promises :)

**Review by wyattjoh — APPROVED** [review]

**Pull request merged**